python/lsst/ts/externalscripts/base_make_calibrations.py

```diff
@@ -1683,26 +1603,138 @@ async def run_block(self):
 
         if len(calib_types):
             for calib_type in calib_types:
-                try:
-                    # Run the pipetask
-                    response_ocps_calib_pipetask = await self.call_pipetask(calib_type)
-                    job_id_calib = response_ocps_calib_pipetask["jobId"]
-                    # Certify the calibrations in self.config.calib_collection
-                    # The quick gain estimation does not need to be certified.
-                    self.log.info(
-                        f"Verification for {calib_type} is not implemented yet "
-                        f"in this script. {calib_type} will be automatically certified."
-                    )
-                    if calib_type != "GAIN":
-                        await self.certify_calib(calib_type, job_id_calib)
+                task = asyncio.create_task(self.process_calibration(calib_type))
+                self.background_tasks.append(task)
+
+        await self.checkpoint("Data-taking part completed.")
 
-                    self.log.info(f"{calib_type} generation job ID: {job_id_calib}")
+        await self.wait_for_background_tasks()
 
-                    # Report the estimated gain from each pair of flats
-                    if calib_type in ["GAIN", "PTC"]:
-                        await self.report_gains_from_flat_pairs(job_id_calib)
-                except Exception:
-                    self.log.exception(f"Error processing {calib_type}. Ignoring...")
+    async def process_images(self, im_type):
+
+        if self.config.generate_calibrations:
+            # 2. Call the calibration pipetask via the OCPS
```

Comment:
Is there a step 1? You might want to remove the comment here. The log message below expains what it is doing.

Reply:
I don't think so, these are comments I left from the original.